### PR TITLE
Improvements in Item Variants

### PR DESCRIPTION
**Pull request by neilLasrado**
- [x] Search Variants added to Item Master
- [x] Added feature for Variants to handle numeric values
- [x] Fixed an issue in filtering Templates in Item List View
- [x] Added Patch to fetch Item Template Attributes
- [x] Allow Editing variant attribute in Variant. Add validation to check duplication.
- [x] Removed Manage Variants
- [x] Validate Change of Attributes in Item Variant
- [x] Test Case

screen shot 2015-08-18 at 1 19 45 pm

Ability to handle Numeric Variants in Attribute Master

screen shot 2015-08-18 at 1 19 12 pm

Make Variant in Item Master

Fixed ##3756

Wont work until https://github.com/frappe/frappe/pull/1249 is Merged

### CODE REVIEW

File: erpnext/patches/v5_4/item_template.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Copyright (c) 2013, Web Notes Technologies Pvt. Ltd. and Contributors
+# License: GNU General Public License v3. See license.txt
+
+from __future__ import unicode_literals
+import frappe
+
+def execute():
```
Comment (anandpdoshi) [inline erpnext/patches/v5_4/item_template.py]: reload `Item`, `Variant Attribute`

**Comment by anandpdoshi** [comment]
To Do:
- [ ] Screenshots
- [ ] Documentation

**Comment by adityaduggal** [comment]
Wow, Finally.

File: erpnext/stock/doctype/item/item.py [hunk 2]
```diff
@@ -458,3 +485,75 @@ def check_stock_uom_with_bin(item, stock_uom):
...
+
+			if (not values.from_range < cint(args[d]) < values.to_range) or ((cint(args[d]) - values.from_range) % values.increment != 0):
+				frappe.throw(_("Attribute value {0} for attribute {1} must be within range of {2} to {3} and in increments of {4}")
+					.format(args[d], d, values.from_range, values.to_range, values.increment))
+		else:
+			if args[d] not in attributes.get(d):
+				frappe.throw(_("Attribute value {0} for attribute {1} does not exist \
+					in Item Attribute Master.").format(args[d], d))
+
+	conds=""
```
Comment (anandpdoshi) [inline erpnext/stock/doctype/item/item.py]: I should not have allowed this code :(

Comment (neilLasrado) [inline erpnext/stock/doctype/item/item.py]: I had changed this .. I guess I didn't commit.

**Pull request merged**